PPOCRLabel.py

```diff
@@ -3665,6 +3679,89 @@ def expandSelectedShape(self):
             self.updateBoxlist()
             self.setDirty()
 
+    def resortBoxposion(self):
+        # Sort from top to bottom, left to right
+        def sort_rectangles(rectangles, row_height_threshold=0.5):
+            if not rectangles:
+                return []
+
+            def get_top_left(rect):
+                xs = [p[0] for p in rect]
+                ys = [p[1] for p in rect]
+                return (min(xs), min(ys))
+
+            avg_height = sum(
+                [
+                    max(p[1] for p in rect) - min(p[1] for p in rect)
+                    for rect in rectangles
+                ]
+            ) / len(rectangles)
+            threshold = avg_height * row_height_threshold
+            indexed_rects = [
+                (i, get_top_left(rect)) for i, rect in enumerate(rectangles)
+            ]
+            indexed_rects.sort(key=lambda x: x[1][1])
+            rows = []
+            current_row = []
+            last_y = indexed_rects[0][1][1]
+            for item in indexed_rects:
+                i, (x, y) = item
+                if abs(y - last_y) <= threshold:
+                    current_row.append(item)
+                else:
+                    rows.append(current_row)
+                    current_row = [item]
+                last_y = y
+            if current_row:
+                rows.append(current_row)
+            sorted_rects = []
+            for row in rows:
+                row.sort(key=lambda x: x[1][0])
+                sorted_rects.extend([rectangles[i] for i, _ in row])
+            return sorted_rects
+
+        # get original elements
+        items = []
+        for i in range(self.BoxList.count()):
+            item = self.BoxList.item(i)
+            items.append({"text": item.text(), "object": item})
+        # get coordinate points
+        rectangles = []
+        for item in items:
+            text = item["text"]
+            try:
+                rect = ast.literal_eval(text)  # 转为列表
+                rectangles.append(rect)
+            except (IndexError, SyntaxError) as e:
```

Comment:
`ast.literal_eval` can raise [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError), [MemoryError](https://docs.python.org/3/library/exceptions.html#MemoryError) and [RecursionError](https://docs.python.org/3/library/exceptions.html#RecursionError), not `IndexError`.

```suggestion
            except (ValueError, SyntaxError) as e:
```